### PR TITLE
api made for project, user and employees

### PR DESCRIPTION
**Pull request by vivekshubhu**

### CODE REVIEW

**Comment by rakeshshubhu** [comment]
remembers commits on present tense

File: app/Project.php [hunk 1]
```diff
@@ -7,4 +7,9 @@
 class Project extends Model
 {
     //
+    public $guarded = [];
+    
+    public function file() {
+        return $this->hasOne('App\File', 'table_id', 'id')->where('table_name', 'projects');
+    }
```
Comment (rakeshshubhu) [inline app/Project.php:14]: moprhs

File: database/migrations/2020_07_21_114006_create_employees_table.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('employees', function (Blueprint $table) {
+            $table->id();
+            $table->string('first_name');
+            $table->string('last_name');
+            $table->string('email');
```
Comment (rakeshshubhu) [inline database/migrations/2020_07_21_114006_create_employees_table.php:20]: nullable middle name

File: database/migrations/2020_07_21_065400_create_files_table.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+     */
+    public function up()
+    {
+        Schema::create('files', function (Blueprint $table) {
+            $table->id();
+            $table->string('name');
+            $table->string('type')->nullable();
+            $table->string('table_name')->nullable();
+            $table->integer('table_id')->nullable();
+            $table->boolean('is_deleted')->default(0);
```
Comment (rakeshshubhu) [inline database/migrations/2020_07_21_065400_create_files_table.php:22]: make this order
```
id
table_id
table_name
name
type
```
do `$table->table_id->index()` on both table_id and table_name

**Comment by rakeshshubhu** [comment]
@vivekshubhu don't push storage images... revert it

**Comment by vivekshubhu** [comment]
> @vivekshubhu don't push storage images... revert it

@rakeshshubhu  revert kasari garne ??

File: app/Employee.php [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace App;
+
+use Illuminate\Database\Eloquent\Model;
+
+class Employee extends Model
+{
```
[the remark this answers is not in the document]
Comment (vivekshubhu) [reply app/Employee.php:8]: yeah it is for staffs,

File: app/Http/Controllers/API/EmployeeController.php [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        if ($request->file) {
+            // dd($employee->file);
+            if ($employee->file) {
+                if ((storage_path('Images/Employees/' . $employee->file->name))) {
+                    unlink(storage_path('Images/Employees/' . $employee->file->name));
+                }
+                $fileName = upload_file($request, 'Images/Employees');
+                $employee->file->update([
+                    'name' => $fileName,
+                ]);
```
[the remark this answers is not in the document]
Comment (vivekshubhu) [reply app/Http/Controllers/API/EmployeeController.php:144]: ahh file remove garxa

**Comment by rakeshshubhu** [comment]
image

chnages in migration file. what changes? be specific if possible.
reverting images means deleting it again and pushing it with gitignore.

**Comment by rakeshshubhu** [comment]
@vivekshubhu yo sabai bhai sakeko cha bhane 
merge

**Pull request merged**